src/python/pants/util/dirutil.py

```diff
@@ -193,7 +193,7 @@ def read_file(filename: str, binary_mode: bool = False) -> Union[bytes, str]:
     return content
 
 
-def safe_walk(path: str, **kwargs: Any) -> Iterator[Tuple[str, List[str], List[str]]]:
+def safe_walk(path: Union[bytes, str], **kwargs: Any) -> Iterator[Tuple[str, List[str], List[str]]]:
```

Comment:
This seems to be one of the only places where we represent a file path as `bytes`?

Reply:
Yes, I was thinking about deprecating `safe_walk()`. It was written in a Python 2 world where mixing bytes and str was standard. 

The behavior is still the case that using `bytes` with `os.walk` will return bytes, so there's an argument for keeping this. But it seems very unlikely this will happen now and not worth the cost of custom code.

--

One frustration with deprecating this, though...depending on `base:deprecated` results in MyPy failing. Going to try to fix that right now - we should be able to depend on deprecrated without it interfering with type hints.